File: raspberry/templog.py
#!/usr/bin/env python
import requests
import os
import datetime

os.system('modprobe w1-gpio')
os.system('modprobe w1-therm')
sensor_path = '/sys/bus/w1/devices/28-0416a4f7ffff/w1_slave'
url = 'http://130.229.163.46:8080/temperatures/'
def tempRead():
    t = open(sensor_path, 'r')
    lines = t.readlines()
    t.close()

    temp_output = lines[1].find('t=')
    if temp_output != -1:
        temp_string = lines[1].strip()[temp_output+2:]
        temp_c = float(temp_string)/1000.0
    return round(temp_c,1)
while True:
    temp = tempRead()
    print temp
    date = datetime.datetime.now().isoformat()
    print date
    query = {'date':date, 'temperature':temp, 'key':'79b7a1a99e4d3799fcfa2fc77ea52782'}
    res = requests.post(url,json=query)
    break

File: raspberry/temp_log.rs
extern crate reqwest;
extern crate chrono;
#[macro_use]
extern crate serde_derive;
extern crate serde;

use std::fs::File;
use std::io::prelude::*;
use chrono::prelude::*;

#[derive(Serialize, Deserialize)]
struct TempReading {
    key: String,
    temperature: f32,
    date: DateTime<Local>
}

fn get_temp() -> f32 {
    let mut f = File::open("/sys/bus/w1/devices/28-0416a4f7ffff/w1_slave")
        .expect("File not found.");
    
    let mut contents = String::new();
    
    f.read_to_string(&mut contents)
        .expect("Could not read the file");
    
    let temp_string = &contents[contents.len()-6..contents.len()-1];
    temp_string.parse::<f32>().unwrap() / 1000.0
    
}

fn post_request(temp: f32, key: String) {
    let local: DateTime<Local> = Local::now();
    let tempreading = TempReading {
        date: local.to_owned(),
        key: key.to_owned(),
        temperature: temp.to_owned()
    };
    let client = reqwest::Client::new();
    let mut response = client.post("http://192.168.2.8:8080/sensors/temperatures/")
        .json(&tempreading)
        .send()
        .expect("Could not send TempReading");
    let mut buf = String::new();
    json.read_to_string(&mut buf).expect("Failed to read response");
    println!("{}", buf);
}

fn main() {
	let temp = get_temp();
	println!("{}", temp);
    let key = String::from("964f55fd39119c18f72192bb92a0971");
	post_request(temp, key);
}
